algorithmic-strategies/divide-and-conquer.md

```diff
@@ -0,0 +1,292 @@
+_Fig. Example run of merge sort showing the index calculations used to split and merge the arrays. Note the convention used of the end index being exclusive._
+
+As we can see in the image above, in the first _divide_ step, the original array of size eight gets divided into two subarrays of size four each. This is done by setting the _starting index_ to _0_, the index of the first element in the array, and the _ending index_ is set to one past the last element in the array, using the convention of the ending index being exclusive. The _midway index_ is then computed using the formula:
+
+<div style="margin:auto;width:fit-content;">
+
+_midway index_ = ⌊(_starting index_ + _ending index_) / 2⌋
+
+</div>
+
+Where ⌊⌋ denotes the mathematical _floor_ operation, or integer truncation.
+
+For the first _divide_ step, the _midway index_ will be _(0+8)/2_ = _4_. In the next _divide_ step, we have two subarrays, one ranging in index from _0_ to _4_ and the other ranging in index from _4_ to _8_. The subarrays are not yet of size one. So, the same action gets repeated to compute the _midway index_. This _divide_ stage continues until the original array of size _n_ is reduced to subarrays of size _1_ each.
+
+A subarray of size one is trivially sorted, since there can be no elements out of place relative to the single value in the array.
+
+The _merge_ stage starts by combining two subarrays at a time. While combining the subarrays containing _7_ and _2_ respectively, the value in each is compared, the smaller value, i.e. _2_, is written to the lower index, and the higher value, i.e. _7_, is written to the higher index. Larger merges would continue comparing the values to be merged element by element, taking the smaller until one subarray is "empty," at which point the remainder of the other array can be copied over.
+
+An auxiliary array of size _n_ is often used to facilitate the merge steps, after which the values are copied from the auxiliary array back to the original array.
+
+The two-way merging continues until there are no more subarrays and the original array is completely sorted.
+
+Since merge sort always divides the list in half at each divide step, there will be _log n_ levels in the division phase. By keeping data in place, and only adjusting indices to keep track of where the subarrays are located, at each level, there are at most _n_ calculations, for a time complexity of the division phase of _O(n log n)_.
```

Comment:
This part transitions out of the algo explanation, and into discussing the complexity.

```suggestion
### Divide and Conquer Complexity in Merge Sort

Since merge sort always divides the list in half at each divide step, there will be _log n_ levels in the division phase. By keeping data in place, and only adjusting indices to keep track of where the subarrays are located, at each level, there are at most _n_ calculations, for a time complexity of the division phase of _O(n log n)_.
```